app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/StartController.scala

```diff
@@ -95,19 +103,49 @@ class StartController @Inject()(
       redirectToIv
   }
 
-  private def buildSubscriptionData(individual: Individual,
-                             maybeBpr: Option[BusinessPartnerRecord])(
+  private def buildTrustSubscriptionData(trust: Trust)(
+    implicit  hc: HeaderCarrier,
+    request: RequestWithSessionDataAndRetrievedData[_]): Future[Result] = {
+    val result =
+      for{
+        bprWithNameAndEmail <- bprService.getBusinessPartnerRecord(Left(trust))
+          .subflatMap[Error,(BusinessPartnerRecord,TrustName,Email)]{ bpr =>
+          (bpr.organisationName, bpr.emailAddress) match {
+            case (None, None) =>
+              Left(Error("Could not find trust name or email address from business partner record"))
+            case (Some(_), None) =>
+              Left(Error("Could not find trust name from business partner record"))
+            case (None, Some(_)) =>
+              Left(Error("Could not find email address from business partner record"))
```

Comment:
we should probably handle this situation like we do for individuals: we should try and get the gg email from auth and if that still doesn't exist then send them down the enter email journey which includes email verification. That can come in a later piece of work though - too much in this PR otherwise

Reply:
I wrote CPDP-571 for this